### PR TITLE
Implement basic sky (sun, moon, stars, atmospheric scattering) + fix matrices

### PR DESCRIPTION
**Pull request by Zanctarian**
originally was just implementing a basic sky, but while working on the sun, I figured out how to get matrices better translated. World effects like bobbing and nausea work now.

### CODE REVIEW

File: rust/wgpu-mc/src/mc/mod.rs [hunk 1]
```diff
@@ -151,9 +153,34 @@ pub struct BlockInstance {
     pub block: MultipartOrMesh,
 }
 
+#[derive(Default)]
```
Comment (birbe) [inline rust/wgpu-mc/src/mc/mod.rs:156]: why not just use `#[derive(Clone)]`?

File: rust/wgpu-mc-jni/src/renderer.rs [hunk 2]
```diff
@@ -104,7 +114,16 @@ pub fn setMatrix(mut env: JNIEnv, _class: JClass, _id: jint, float_array: JFloat
     }
 
     let slice_4x4: [[f32; 4]; 4] = *bytemuck::from_bytes(bytemuck::cast_slice(&converted));
-    MATRICES.lock().projection = slice_4x4;
+
+    if id == 0 {
```
Comment (birbe) [inline rust/wgpu-mc-jni/src/renderer.rs:118]: use a match brace

**Review by birbe — CHANGES_REQUESTED** [review]

File: rust/wgpu-mc/src/mc/mod.rs (hunk 1 above)
Comment (Zanctarian) [reply rust/wgpu-mc/src/mc/mod.rs:156]: wont let me because of the hashmap

File: rust/wgpu-mc-jni/src/renderer.rs (hunk 2 above)
Comment (Zanctarian) [reply rust/wgpu-mc-jni/src/renderer.rs]: 👍

**Comment by Zanctarian** [comment]
note: vecs don't seem to work with push constants.  They get inverted and the shaders don't read the right spot in memory. Hopefully the graph rework on the next wgpu major version will allow vec push constants. As a temporary workaround, I just passed each axis as a f32.

other than that, tested and ready to merge after review

**Pull request merged**